### ISSUE TITLE
s6-mkdir: warning: unable to mkdir /var/run/s6: Permission denied

### ISSUE THREAD

**Issue by Ashtonian** [linuxserver/docker-nextcloud#98]
new image getting permission issue: get the following `s6-mkdir: warning: unable to mkdir /var/run/s6: Permission denied` 
```yml
 nextcloud:
    logging: *default-logging
    image: linuxserver/nextcloud
    container_name: nextcloud
    environment:
      PUID: 5001
      PGID: 5001
      TZ: America/Chicago
      NEXTCLOUD_TRUSTED_DOMAINS: # not sure gets passed in 
      NEXTCLOUD_DATA_DIR: /data # not sure gets passed in 
    volumes:
      - nextcloud:/var/www/html
      - nexctcloud_config:/var/www/html/config
      - nexctcloud_config2:/config # not sure if the same as /var/www/html/config
      - nexctcloud_custom_apps:/var/www/html/custom_apps
      - nextcloud_data:/data
      - nextcloud_media:/ncmedia
    # ports:
    #   - 443:80
    restart: unless-stopped
    networks:
      - nextcloud
      - public
    labels:
      traefik.enable: "true"
      traefik.docker.network: "public"
      traefik.backend: "nextcloud"
      traefik.frontend.rule: "Host:"
      traefik.port: "80"
      traefik.protocol: "http"
    user: "5001:5001"
```

Ubuntu 18.04

```
s6-mkdir: warning: unable to mkdir /var/run/s6: Permission denied
s6-mkdir: warning: unable to mkdir /var/run/s6: Permission denied

```

**Comment by thelamer** [comment]
S6 runs as root you do not run as a user. 

Drop ```    user: "5001:5001"``` from your compose file.

**Comment by ahmetkca** [comment]
Can we run S6 as non-root?

**Comment by aptalca** [comment]
The supervisor we use, S6, and our init steps require root. The apps drop privileges after init and run as the PUID/PGID set (most of them).

**Comment by ahmetkca** [comment]
@aptalca  How am I suppose deploy this to production. For example Heroku runs the Dockerfile as non-root. How can I use it in that case?

**Comment by aptalca** [comment]
That's a question for Heroku. Our containers require root.

**Comment by bernardgut** [comment]
@aptalca running any linuxserver container anywhere else than at home (e.g. production) is next to impossible because of this. Most cloud providers don't allow root containers. Rightfully so.

**Comment by thespad** [comment]
Quite the necrothread.

TL;DR almost all our containers will happily run rootless. They will not run as a non-root user. If your platform of choice requires running as a non-root user and does not support rootless operation then our images will not run on it and that's unlikely to change in the near future for a number of different reasons.

**Comment by bernardgut** [comment]
Quite the necrothread indeed. 

> almost all our containers will happily run rootless. 

This isn't true. Modern kubernetes (1.30+) is turning seccompProfiles on by default and mendating users to add `securityProfiles` to all pods/containers workload. 

Last time I checked (Yesterday) Running any of your images with the most basic :

```
    spec:
      securityContext:
        runAsNonRoot: true
        seccompProfile:
          type: RuntimeDefault
      ...
      containers:
      - name: controller-manager
        securityContext:
          allowPrivilegeEscalation: false
          capabilities:
            drop:
            - ALL
        ...
```

will not work.
 
You can test this yourself on Minikube. It takes 10 minutes. Either way I did not want to argue. I just provide the info and let you do whatever you want. 

For anyone else landing on this thread:  If you need to run on Kubernetes 1.30+ as non-root, You can use https://github.com/onedr0p/containers images as they adhere to KISS, have one process per container and no init-system requiring root. I was bored so I even built a helm chart around those to deploy them : https://github.com/sekops-ch/k8s-mediamanager

Cheers
B.

**Comment by Roxedus** [comment]
Kubernetes is not a platform we target. https://docs.linuxserver.io/misc/support-policy

**Comment by thespad** [comment]
`runAsNonRoot` is not rootless, it's non-root. 

Rootless operation, as in the context of docker and podman, runs the entire container environment in its own namespace with userns remapping and we provide limited support for it.

Again, running as a non-root user will not work as it stands, regardless of other environmental configuration.
